### PR TITLE
D7 Temp paths

### PR DESCRIPTION
**Pull request by alexfornuto**
Closes: #4972 

## Summary

**[Temporary File Management Doc Update](https://pantheon.io/docs/tmp)** - Adjusted example code to overwrite D7 temp paths to account for Pressflow settings.

## Post Launch

- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5732 | Difference |
| --- | --- | --- | --- |
| Performance Score | `43` | `47` | `4` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5732` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/21151/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/21151/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by greg-1-anderson** [comment]
Setting the temporary path to /tmp is fine on COS, but it should remain /srv/bindings/BDIR/tmp on heirloom. The way to find the temporary path on Pantheon would be "$HOME/tmp", but the file temporary path should already be set to this location, and setting it to any other directory would be undesirable.

Is therefore should never be necessary to override file_temporary_path on Pantheon. If customers want to set the file_temporary_path on their local environment, not, then perhaps these examples should be wrapped in conditionals so that they do not run on Pantheon. If we're not running on Pantheon, though, then there won't be any pressflow settings to change.

What is the problem this documentation update is trying to solve?

**Comment by alexfornuto** [comment]
> What is the problem this documentation update is trying to solve?

@jfussion can you provide more context for this?

**Comment by jfussion** [comment]
@greg-1-anderson for D7 site, it is not possible to change the temp path via settings.php and requires to modify the pressflow config as mentioned in #4972. Changing the temp dir is necessary when trying to achieve persistent temp dir https://pantheon.io/docs/tmp#persistent-temporary-path-workaround

File: source/content/tmp.md [hunk 1]
```diff
@@ -16,18 +17,31 @@ export env=dev
...
+  $pressflow = json_decode($_SERVER['PRESSFLOW_SETTINGS'], TRUE);
+
+  $pressflow['conf']['file_temporary_path'] = $pressflow['conf']['file_directory_temp'] = '/tmp';
+
+  $_SERVER['PRESSFLOW_SETTINGS'] = json_encode($pressflow);
+
 }
 ```
+
+Change `sites/default/files/private/tmp` to your desired temporary file path.
```
Comment (greg-1-anderson) [inline source/content/tmp.md:41]: I didn't notice the recommendation here to put the private files directory in Valhalla; note that the code snippet above is inconsistent with this advice, recommending `/tmp` instead.

**Comment by alexfornuto** [comment]
@jfussion does setting the temp directory for the specific module work for the instances that spawned this request?

@greg-1-anderson what do you think about the [current first section](https://pantheon.io/docs/tmp#default-temporary-path) of this doc? It sounds like Joseph is saying in the issue that it doesn't work, and you're saying you shouldn't change it anyway.

**Comment by jfussion** [comment]
@alexfornuto yes, tmp dir for specific module/plugin works but there are some modules and plugins that doesn't have any config for tmp dir and therefore lots they end up changing the global tmp dir.

> The persistent temporary path workaround recommends setting the temporary directory for just one plugin / module, using a variable specific to that plugin or module. I think it's not a good practice to put all temporary files in valhalla. I am -1 on recommending this in documentation. I am also -1 on recommending that folks rewrite Pressflow Settings.
> 
> If you wish to move forward with this PR, please get approval from product. I recommend closing without merge.

This completely make sense and if we go for this, we may just remove the first section, Default Temporary Path, while correct, it gives the impression to the customer that it is ok to have the tmp dir in valhalla.

**Comment by alexfornuto** [comment]
OK. Sounds like we're in agreement on closing this, and possible different changes to this page. Regarding the latter, I'd like to see a new issue and/or PR for that, as this thread is a bit too convoluted at this point to keep track of what anyone is actually suggesting.

**Pull request closed without merging**